Rules/AvoidGlobalAliases.cs

```diff
@@ -0,0 +1,195 @@
+
+                    if ((nextAst is StringConstantExpressionAst)   // if the cmdlet looks like -Scope Global
+                        && (((StringConstantExpressionAst)nextAst).Value.ToString().Equals("Global", StringComparison.OrdinalIgnoreCase)))
+                    {
+                        records.Add(new DiagnosticRecord(
+                                string.Format(CultureInfo.CurrentCulture, Strings.AvoidGlobalAliasesError),
+                                ((StringConstantExpressionAst)nextAst).Extent,
+                                GetName(),
+                                DiagnosticSeverity.Warning,
+                                fileName,
+                                ((StringConstantExpressionAst)nextAst).Value));
+                    }
+                }
+            }
+
+            return AstVisitAction.SkipChildren;
+        }
+        #endregion
+
+        private Ast FindNextAst(Ast ast)
+        {
+            IEnumerable<Ast> matchingLevelAsts = ast.Parent.FindAll(item => item is Ast, true);
+
+            Ast currentClosest = null;
+            foreach (var matchingLevelAst in matchingLevelAsts)
+            {
+                if (currentClosest == null)
+                {
+                    if (IsAstAfter(ast, matchingLevelAst))
+                    {
+                        currentClosest = matchingLevelAst;
+                    }
+                }
+                else
+                {
+                    if ((IsAstAfter(ast, matchingLevelAst)) && (IsAstAfter(matchingLevelAst, currentClosest)))
+                    {
+                        currentClosest = matchingLevelAst;
+                    }
+                }
+            }
+
+            return currentClosest;
+        }
+
+        private bool IsAstAfter(Ast ast1, Ast ast2)
+        {
+            if (ast1.Extent.EndLineNumber > ast2.Extent.StartLineNumber)  // ast1 ends on a line after ast2 starts
+            {
+                return false;
+            }
+            else if (ast1.Extent.EndLineNumber == ast2.Extent.StartLineNumber)
+            {
+                if (ast2.Extent.StartColumnNumber > ast1.Extent.EndColumnNumber)
+                {
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            else   // ast2 starts on a line after ast 1 ends
+            {
+                return true;
+            }
+        }
+
+        private bool IsScopeParameterForNewAliasCmdlet(CommandParameterAst commandParameterAst)
+        {
+            if (commandParameterAst == null || commandParameterAst.ParameterName == null)
+            {
+                return false;
+            }
+
+            if (commandParameterAst.ParameterName.Equals("Scope", StringComparison.OrdinalIgnoreCase)
+                && (commandParameterAst.Parent is CommandAst)
+                && IsNewAliasCmdlet((CommandAst)commandParameterAst.Parent))
```

Comment:
You are checking for `IsNewAliasCmdlet` in VisitCommand implementation too. Since you skip its children if the command is not an alias, your `VisitCommandParameter` won't be called if the command is not `new-alias` or `nal` and hence these two checks - '*Parent. is commandast' and `IsNewAliasCmdlet' are redundant.